solvers/skglm.py

```diff
@@ -28,22 +30,19 @@ def set_objective(self, X, y, lmbd):
         n_samples = self.X.shape[0]
 
         warnings.filterwarnings('ignore', category=ConvergenceWarning)
-        self.logreg = SparseLogisticRegression(
-            alpha=self.lmbd / n_samples, max_iter=1, max_epochs=50_000,
-            tol=1e-12, fit_intercept=False, warm_start=False, verbose=False)
+        self.estimator = GeneralizedLinearEstimator(
+            datafit=Logistic(),
+            penalty=L1(lmbd / n_samples),
+            solver=ProxNewton(tol=1e-12, fit_intercept=False)
+        )
 
         # Cache Numba compilation
-        self.run(1)
+        self.run(5)
```

Comment:
the zig-zag pattern of the objective that we noticed last time was due to numba jit-compilation

Reply:
Is 2 enough instead of 5?

Reply:
I've just checked, and this behavior persists for ``n_iter <= 4``.

Reply:
it may mean that in the first 3 iters, we don't enter some part of the code. It feels like it should investigated more instead of being being fix with this 5 value that somewhat reads like an arbitrary one.

Reply:
Nothing suspicious when running the solver in verbose mode.
I pushed a ``debug_script`` for that

Reply:
It's not trivial that 5 iterations are needed and 1 is not enough. Can you add a comment so that somebody doesn't change this back to 1 or 2 in the future?